src/chapters/auth/login-registration-forms.rst

```diff
@@ -0,0 +1,378 @@
+--------------------------
+
+Before creating handler methods for rendering and processing our login and registration forms, we need some utility methods for working with sessions. Below ``userRepository``, let's add the following class members:
+
+.. sourcecode:: java
+   :lineno-start: 27
+
+   private static final String userSessionKey = "user";
+
+   public User getUserFromSession(HttpSession session) {
+      Integer userId = (Integer) session.getAttribute(userSessionKey);
+      if (userId == null) {
+         return null;
+      }
+
+      Optional<User> user = userRepository.findById(userId);
+
+      if (user.isEmpty()) {
+         return null;
+      }
+
+      return user.get();
+   }
+
+   private static void setUserInSession(HttpSession session, User user) {
+      session.setAttribute(userSessionKey, user.getId());
+   }
+
+This code allows us to store and retrieve the login status of a user in a session. More specifically, a logged-in user's user ID will be stored in their session. 
+
+.. list-table:: Sample session data for a logged-in user
+   :stub-columns: 1
+
+   * - session_id
+     - 81LfWG9
+   * - user
+     - 42
+
+The static field ``userSessionKey`` is the key used to store user IDs. ``setUserInSession`` uses an ``HttpSession`` object (part of the standard ``javax.servlet.http`` package) to store key/value pair. ``getUserFromSession`` looks for data with the key ``user`` in the user's session. If it finds one, it attempts to retrieve the corresponding ``User`` object from the database. If no user ID is in the session, or if there is no user with the given ID, ``null`` is returned.
+
+.. admonition:: Note
+
+   The ``HttpSession`` class handles the details of session creation and lookup for us, including generating unique session IDs and session cookies.
+
+These utility methods will allow our handlers to manage authentication.
+
+Form DTOs
+---------
+
+Our login and registration forms will use DTOs to help with form rendering and processing. Furthermore, since these forms will be similar---both require a username and password---we'll use inheritance in creating our DTOs.
+
+The DTO for the login form needs only ``username`` and ``password`` fields.
+
+.. sourcecode:: java
+   :lineno-start: 10
+
+   public class LoginFormDTO {
+
+      @NotNull
+      @NotBlank
+      @Size(min = 3, max = 20, message = "Invalid username. Must be between 3 and 30 characters.")
+      private String username;
+
+      @NotNull
+      @NotBlank
+      @Size(min = 4, max = 20, message = "Invalid password. Must be between 5 and 30 characters.")
+      private String password;
+
+      public String getUsername() {
+         return username;
+      }
+
+      public void setUsername(String username) {
+         this.username = username;
+      }
+
+      public String getPassword() {
+         return password;
+      }
+
+      public void setPassword(String password) {
+         this.password = password;
+      }
+
```

Comment:
I'm a little lost on what values these fields are for - are they the values from the form fields?

Reply:
actually, i just re-read the tip below and answered my own question - but we may find it helpful to call this idea out more specifically for this instance because it is right after drilling home the idea to not store password values

Reply:
I added a note about the `password` field. Let me know what you think.

Reply:
yeah thats great - although you should add a comma after the first clause